### PR TITLE
Fix excerpts in new Storage

### LINKED ISSUE #7602: [3.6] `record.excerpt` doesn't work in new storage.
To reproduce, go to `/showcases` or `/search?search=ipsum` on a stock Bolt 3.6 setup. The `listing.twig` outputs content using `{{ record.excerpt }}`, only nothing shows. 

Output on 3.6: 

screen shot 2018-08-12 at 18 41 10

Output on 3.5: 

screen shot 2018-08-12 at 18 41 24

### PR DESCRIPTION
**Pull request by bobdenotter**
Fixes #7602.

Additionally, prevents excerpts from looking 'ugly' with serialised JSON in them.

### CODE REVIEW

File: src/Helpers/Excerpt.php [hunk 1]
```diff
@@ -73,8 +74,14 @@ public function getExcerpt($length = 200, $includeTitle = false, $focus = null)
...
             array_walk($this->body, function ($value, $key) use (&$excerpt, $stripKeys) {
-                if (is_string($value) && !in_array($key, $stripKeys)) {
-                    $excerpt .= $value . ' ';
+                if (in_array($key, $stripKeys)) {
+                    return;
+                }
+                // We need non-empty strings that don't look like serialized JSON.
+                // Otherwise, Twig Markup is also OK.
+                if ((is_string($value) && !empty($value) && ($value[0]!='{')) ||
+                    (is_object($value) && ($value instanceof Markup))) {
```
Comment (SvanteRichter) [inline src/Helpers/Excerpt.php]: is_object is not needed:

> instanceof does not throw any error if the variable being tested is not an object, it simply returns FALSE.

File: src/Helpers/Excerpt.php [hunk 2]
```diff
@@ -73,8 +74,14 @@ public function getExcerpt($length = 200, $includeTitle = false, $focus = null)
...
             $excerpt = '';
             array_walk($this->body, function ($value, $key) use (&$excerpt, $stripKeys) {
-                if (is_string($value) && !in_array($key, $stripKeys)) {
-                    $excerpt .= $value . ' ';
+                if (in_array($key, $stripKeys)) {
+                    return;
+                }
+                // We need non-empty strings that don't look like serialized JSON.
+                // Otherwise, Twig Markup is also OK.
+                if ((is_string($value) && !empty($value) && ($value[0]!='{')) ||
```
Comment (SvanteRichter) [inline src/Helpers/Excerpt.php]: Perhaps we should also check for `($value[0] !== '[')` (some extensions store json arrays IIRC), and strict comparison would be nice even though it probably won't make a difference.

**Review by SvanteRichter — DISMISSED** [review]

File: src/Helpers/Excerpt.php (hunk 2 above)
Comment (bobdenotter) [reply src/Helpers/Excerpt.php]: Good points both. Will do.

File: src/Helpers/Excerpt.php (hunk 1 above)
Comment (bobdenotter) [reply src/Helpers/Excerpt.php]: Ack.

**Comment by bobdenotter** [comment]
Updated.

**Review by SvanteRichter — APPROVED** [review]

**Pull request merged**